qa/workunits/windows/test_rbd_wnbd.py

```diff
@@ -374,20 +411,26 @@ def __init__(self,
         self.bs = bs
         self.iodepth = iodepth
         self.op = op
+        if op not in ("read", "randread"):
+            self.requires_disk_write = True
```

Comment:
Hrm, this seems fishy to me -- if I'm not mistaken, this would essentially turn a static field into an instance field.  Why `requires_disk_write` and `requires_disk_online` are static fields to begin with?

Reply:
You clearly want each class to have its own pair of these values so static fields don't make sense IMO.  I'd just make each subclass of `RbdTest` to explicitly set `self.requires_disk_write` and `self.requires_disk_online` in `__init__`.

Alternatively, if you really want to provide defaults, you could use `getattr` in `RbdTest::initialize()`.

Reply:
Python allows object attributes to override class attributes and this is one of the common use cases. Here's a sample:

```
class A(object):
	a = 10

	def override_a(self, val):
		self.a = val

obj = A()

class B(A):
	a = 30

print("A.a: ", A.a)
print("obj.a: ", obj.a)

print("overriding obj.a with 20")
obj.override_a(20)

print("A.a: ", A.a)
print("obj.a: ", obj.a)
print("B().a: ", B().a)
```

Output:
```
A.a:  10
obj.a:  10
overriding obj.a with 20
A.a:  10
obj.a:  20
B().a:  30
```

IMO this is a very simple and convenient approach but if you strongly prefer, I can drop the class attributes and rely on the constructors.

The rationale behind behind using a class attribute is that based on the class type, we may know whether the test is going to perform write operations or not. In other cases, we may determine based on the test parameters (e.g. fio operation type), in which case the test is simply overriding the class default.

Reply:
I know that Python allows this, it just seems weird to me.  I don't have a strong opinion and won't insist especially if you say it's actually convenient here.